Userland/Libraries/LibJS/Runtime/PromiseCapability.cpp

```diff
@@ -31,6 +31,22 @@ void PromiseCapability::visit_edges(Cell::Visitor& visitor)
     visitor.visit(m_reject);
 }
 
+namespace {
+struct ResolvingFunctions final : public Cell {
+    JS_CELL(ResolvingFunctions, Cell);
+
+    Value resolve { js_undefined() };
+    Value reject { js_undefined() };
```

Comment:
Let's type this slightly stricter as `GCPtr<FunctionObject>` - `undefined` in the spec is not always to be taken literally but can be represented as any C++ null state.

`visit_edges()` should not need any changes for this, only the `is_undefined()` checks below

Reply:
> `GCPtr<FunctionObject>`

The test fix part of this PR comes from the fact that we allow not just functions to be assigned to `resolve`/`reject`. Take this test from [test/built-ins/Promise/all/capability-executor-called-twice.js](https://github.com/tc39/test262/blob/main/test/built-ins/Promise/all/capability-executor-called-twice.js#L88C1-L102C87):

```js
checkPoint = "";
function fn5(executor) {
  checkPoint += "a";
  executor("invalid value", 123);
  checkPoint += "b";
  executor(function() {}, function() {});
  checkPoint += "c";
}
Object.defineProperty(fn5, 'resolve', {
  get() { throw new Test262Error; }
});
assert.throws(TypeError, function() {
  Promise.all.call(fn5, []);
}, "executor initially called with (String, Number)");
assert.sameValue(checkPoint, "ab", "executor initially called with (String, Number)");
```

If the pointer could only contain a `FunctionObject`, we'd have to skip assigning the string to `resolve` and the number to `reject` in the first call to `executor`. But then, those would stay null/undefined on the second (correct) call to `executor`, which would then succeed; so in the end, we'd observe the wrong value (`"abc"`) in `checkPoint`.

Reply:
oh, I totally missed this! seems good then :)